src/map.rs

```diff
@@ -937,6 +937,74 @@ where
         n.leading_zeros() as isize | (1 << (RESIZE_STAMP_BITS - 1)) as isize
     }
 
+    // TODO
+    fn try_presize<'g>(&self, size: usize, guard: &'g Guard) {
+        let new_capacity = if size >= MAXIMUM_CAPACITY / 2 {
+            MAXIMUM_CAPACITY
+        } else {
+            // 1.5 * size + 1
+            let size = size + (size >> 1) + 1;
+
+            std::cmp::min(MAXIMUM_CAPACITY, size.next_power_of_two())
+        };
+
+        let c = new_capacity as isize; // TODO
+
+        loop {
+            let sc = self.size_ctl.load(Ordering::SeqCst);
+            if sc < 0 {
+                break;
+            }
+
+            let table = self.table.load(Ordering::SeqCst, &guard);
+
+            // TODO: clean this up?
```

Comment:
I propose using `0` instead of `None`. That way the check below can simply be `if n == 0`. You could possibly simplify it further with:
```rust
let mut n = 0;
if !table.is_null() {
    // safety: table was read under guard, which still lives, so table cannot be dropped this epoch
    n = unsafe { table.deref() }.len();
}
```

Reply:
That's a great idea, what do you think of something like this:
```rust
let n = match table.is_null() {
    true => 0,
    false => unsafe { table.deref() }.len(),
};
```

Reply:
That way we wouldn't need the `mut n`, even though we don't really mutate it

Reply:
Beautiful!